core/arch/arm/include/arm64.h

```diff
@@ -304,6 +304,37 @@ DEFINE_U64_REG_WRITE_FUNC(mair_el1)
 
 DEFINE_REG_READ_FUNC_(cntpct, uint64_t, cntpct_el0)
 
+/* Register read/write functions for GICC registers by using system interface */
+static inline uint32_t read_icc_ctlr(void)
```

Comment:
You're right. I just noticed DEFINE_U32_REG_XXX_FUNC.
Seems for aarch32, we don't have this kind of macros to define the read/write functions of system registers?